### ISSUE TITLE
How to access to rutorrent outside of my house ?

### ISSUE THREAD

**Issue by ztheoz** [binhex/arch-rtorrentvpn#61]
Hi,
because of the VPN I can't access to rutorrent outside my local network, even if I open the 9080 port on my route (I'm having a timeout)r.
I think that the solution is either to redirect the port 1000 to 127.0.0.1:9080 or to make a reverse proxy with ngnix to point #domain#/rutorrent to 127.0.0.1:9080.
But I don't know how to do so, that's why I'm asking help.
Thanks !

**Comment by binhex** [comment]
Check the /config/supervisord.log it will tell you the problem, it's probably related to iptable mangle module not present.

**Comment by GottZ** [comment]
i literarily run a nginx on the host system and proxy forward into the correct ports.

```
root@tesla ~ # cat /etc/nginx/sites-enabled/rutorrent
server {
        server_name rutorrent.home.example.com;

        charset UTF-8;
        default_type application/octet-stream;
        include mime.types;
        client_max_body_size 200M;
        #include errorpages;
        #include sslstrict;

        location / {
                proxy_set_header Host $host;
                proxy_redirect off;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection "upgrade";
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header X-Forwarded-Ssl on;
                proxy_set_header Authorization "Basic YWRtaW46cnV0b3JyZW50";
                proxy_pass http://localhost:9080;
        }

        listen 443 ssl; # managed by Certbot
        ssl_certificate /etc/letsencrypt/live/home.example.com/fullchain.pem; # managed by Certbot
        ssl_certificate_key /etc/letsencrypt/live/home.example.com/privkey.pem; # managed by Certbot
        include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
        ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

        auth_request "/__auth";

        location = /__auth {
                internal;
                proxy_pass http://localhost:8081;
                proxy_pass_request_body off;
                proxy_set_header Host $host;
                proxy_set_header Content-Length "";
                proxy_set_header X-Original_host $host;
                proxy_set_header X-Original_path $request_uri;
        }

        error_page 401 = @error401;
        location @error401 {
                return 302 https://login.home.example.com?host=$host&path=$request_uri;
        }
}
server {
    if ($host = rutorrent.home.example.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

        listen 80;
        server_name rutorrent.home.example.com;
    return 404; # managed by Certbot
}
```

in my docker configuration i simply bound 9080 on localhost of the host system.

that `auth_request "/__auth";` and `location = /__auth {` stuff is simply my own login system in front of it.
you could just remove it aswell as remove `proxy_set_header Authorization "Basic YWRtaW46cnV0b3JyZW50";` so this way you will be asked for the rutorrent username and password.

**Comment by GottZ** [comment]
yep. i can tell you why you get a 404. it's simply because `location ^~ /rutorrent` will forward your request to `http://localhost:9080/rutorrent` instead of `http://localhost:9080`

you should probably avoid doing that and use a proper subdomain. well there is some way to magically rewrite these within nginx but luckily i never had to worry about that before.

https://serverfault.com/questions/379675/nginx-reverse-proxy-url-rewrite

i cannot say that this will work since it requires relative rather than absolute paths within the web application. it does not rewrite what rutorrent will send to your browser. so it could potentially break rutorrent. but we never know until you try.

**Comment by GottZ** [comment]
https://serverfault.com/a/725433/166248

just realized this..

feel free to play around with it. maybe you get it running.

**Comment by ztheoz** [comment]
Ok, thanks.
I've added a "/" after proxy_pass http://localhost:9080, and it almost worked : i've seen that I was connected to rutorrent, but it was broken (I've added a screenshot).
So I tried with the https version, by replacing http://localhost:9080 with https://localhost:9443, but I've got the same result :
![rutorrent_bug](https://user-images.githubusercontent.com/32197190/43048991-c5e39ea4-8df0-11e8-93ef-5082409471d9.png) 
I'm going to fix that, but I don't know what to change...
If I can't solve that I'll try with a subdomain.
